cetcd.c

```diff
@@ -63,7 +64,13 @@ void cetcd_client_init(cetcd_client *cli, cetcd_array *addresses) {
 
     addrs = cetcd_array_create(cetcd_array_size(addresses));
     for (i=0; i<cetcd_array_size(addresses); ++i) {
-        cetcd_array_append(addrs, sdsnew(cetcd_array_get(addresses, i)));
+        addr = sdsnew(cetcd_array_get(addresses, i));
+        if ( strncmp(addr, "http", 4)) {
+            a = addr;
+            addr = sdscatprintf(sdsempty(), "http://%s", a);
+            sdsfree(a);
```

Comment:
The logic here is complex and can be avoided. You allocate a string again when the if condition is true and free it with a extra variable `a`. There is a more simpler way: Allocate one time just when you append the address.